### PR TITLE
build: Add Temporal to docker-compose.yml and fix host.docker.internal

### PR DESCRIPTION
**Pull request by LuedTek**
## Description & Overview
Docker compose extension that sets up temporal. Installation guide should be updated to include the command ```-f docker-compose.temporal.yml``` for prod and ```-f docker-compose.temporal.yml -f docker-compose.temporal.dev.yml``` for dev.

Documentation involving the installation changes have not been made in this PR. 

## Related Tickets & Documents
To run: 

prod:
```docker compose -f docker-compose.yml -f docker-compose.temporal.yml up```

dev: 
```docker compose -f docker-compose.yml -f -f docker-compose.dev.yml -f docker-compose.temporal.yml -f docker-compose.temporal.dev.yml up```

### CODE REVIEW

**Review by topher-lo — COMMENTED** [review]
WIP

**Comment by daryllimyt** [comment]
# Final changes
- Add Temporal services into Tracecat's docker compose. You can now run `docker compose up` to spin up a fully working instance.
- Removes `host.docker.internal` as all services now communicate using docker's internal dns
- Set Temporal services to log level `warn`
- Update conftests
- Remove extraneous compose extensions
- Prefix all tracecat services with 'tracecat'
- Update docker networks names


# Remarks
- Tracecat's worker will traceback with errors until it connects to the Temporal server.

# Testing
- All unit tests passing
- Able to open up UI and create workflow
- Able to use CLI to run `commit`, `up`, and `run` with stress_adder_test.yml
- Verified that worker ran in logs, verified that workflow ran in temporal UI
- Able to see the adder tree graph in the UI after commit

File: .env.example [hunk 1]
```diff
@@ -23,9 +24,9 @@ TRACECAT__API_URL=http://api:8000
 TRACECAT__PUBLIC_RUNNER_URL=http://localhost:8000
 
 # --- Postgres ---
-POSTGRES_USER=postgres
-POSTGRES_PASSWORD=postgres
-TRACECAT__DB_URI=postgresql+psycopg://${POSTGRES_USER}:${POSTGRES_PASSWORD}@host.docker.internal:5432/postgres
+TRACECAT__POSTGRES_USER=postgres
+TRACECAT__POSTGRES_PASSWORD=postgres
+TRACECAT__DB_URI=postgresql+psycopg://${TRACECAT__POSTGRES_USER}:${TRACECAT__POSTGRES_PASSWORD}@host.docker.internal:5432/postgres
```
Comment (topher-lo) [inline .env.example]: Please fix?

**Review by topher-lo — CHANGES_REQUESTED** [review]
postgres uri still has docker internal

**Comment by daryllimyt** [comment]
Nice catch, fixed + reran tests

**Review by topher-lo — APPROVED** [review]
LGTM

**Pull request merged**